iFixit/react-commerce
# Merge hero sections components into one

I'm trying to divide the work being done in #265 in small chunks that are easier to review. This PR aligns the hero section component to how sections are built in product page template and page template.

## QA

1. Open a [product list without hero image](https://react-commerce-git-refactor-product-list-hero-section-ifixit.vercel.app/Parts) and verify that the hero section looks the same of the [equivalent product list in production](https://www.ifixit.com/Parts) (saved for content differences in Strapi)
2. Open a [product list with hero image](https://react-commerce-git-refactor-product-list-hero-section-ifixit.vercel.app/Parts/Nokia_G22) and verify that the hero section looks the same of the [equivalent product list in production](https://www.ifixit.com/Parts/Nokia_G22) (saved for content differences in Strapi)
